src/ngx_http_lua_bodyfilterby.c

```diff
@@ -140,6 +146,19 @@ ngx_http_lua_body_filter_by_chunk(lua_State *L, ngx_http_request_t *r,
 
     lua_settop(L, 0);
 
+    if (ctx->body_exited) {
+        /* send last content */
+        for (cl = in; cl->next; cl = cl->next) {
+            /* do nothing */
+        }
+        /* make ngx_http_write_filter_module happy */
+        cl->buf->last_buf = 1;
+
+        ctx->eof = 1;
+
+        return NGX_OK;
```

Comment:
@agentzh  I agree with your opinion that send a broken response body is wrong. 
We just want to exit to bodyfilter and stop to send remaining response body while we get some problem in `bodyfilter` phase. But in this phase response header maybe has sent. How can we deal with status like this?
Sorry of my poor English, most of them is come from google's translation :(